install.sh

```diff
@@ -996,7 +996,7 @@ esac
 
 cat <<EOS
 - Run these two commands in your terminal to add Homebrew to your ${tty_bold}PATH${tty_reset}:
-    (echo; echo 'eval "\$(${HOMEBREW_PREFIX}/bin/brew shellenv)"') >> ${shell_profile}
+    grep -qsF 'eval "\$(${HOMEBREW_PREFIX}/bin/brew shellenv)"' ${shell_profile} || (echo; echo 'eval "\$(${HOMEBREW_PREFIX}/bin/brew shellenv)"') >> ${shell_profile}
```

Comment:
This makes the preceding line potentially incorrect/confusing, in that it’ll say “Run these **two** commands” while showing one. Also, command flags should be expanded in scripts so they’re easier to understand: `--quiet --no-messages --fixed-strings`.

Reply:
> This makes the preceding line potentially incorrect/confusing, in that it’ll say “Run these two commands” while showing one.

Could you clarify what you mean?

> Also, command flags should be expanded in scripts so they’re easier to understand: --quiet --no-messages --fixed-strings

Good pro-tip. I will use the full flag names.

Reply:
> Could you clarify what you mean?

See the line before your code. I says:

```shell
- Run these two commands in your terminal to add Homebrew to your ${tty_bold}PATH${tty_reset}:
```

**Two** commands, it says. However, what you change does is make it so one of the two commands does not display in a specific condition (already having the `eval` line). Which means that users who meet that condition will be told to run two commands but only see **one**.

That is a possible source of confusion, which leads to wasted time, which leads to frustration, which leads to the Dark Side. Er, I mean it leads to avoidable bug reports.

Reply:
Even if the `eval` line is present in the user's shell profile, the command would still display regardless of the boolean returned by the grep. This is because the two lines are not evaluated; they are just printed to stdout in using a `cat` heredoc.

Reply:
I think @vitorgalvao means you should instead:
1. _parameterize_  "these two commands" and the `brew shellenv` injection string, so that they can be replaced with "this command" and the empty string respectively if the `brew shellenv` line already exists, and
2. test for `brew shellenv` existence _before_ the `cat`, instead of leaving it to users to run the test themselves

The second step above is _critical_; it's to be expected that some users will type in commands manually instead of copy-n-pasting, so complex command lines like what you're proposing are not a good idea.

There's also no technical reason for the installer to avoid doing the test on the user's behalf. The probability of a `brew shellenv` magically appearing in a user's profile, between the time they see those commands and actually run them, is essentially zero.